params/params.py

```diff
@@ -3,34 +3,118 @@
 
 Model parameters
 ================
+Defines classes for the user-facing parameters of the model.
+"""
 
-``Params``
-----------
+from pydantic import Field, PositiveInt, PositiveFloat, confloat
 
-All parameters of the model.
+from dash_models import DashEditorPage
+from .interpolated_param import InterpolatedParam
 
-This is not completely generalized,
-in the sense that new parameters need to be added manually.
 
-----
-"""
+class EmissionsPricing(DashEditorPage):
+    """
+    Pricing for an emissions tax, aka a carbon tax
+    """
+    CO2: InterpolatedParam = Field(InterpolatedParam(), title="CO2 Pricing (ILS/ton)")
+    SOX: InterpolatedParam = Field(InterpolatedParam(), title="SOX Pricing (ILS/ton)")
+    NOX: InterpolatedParam = Field(InterpolatedParam(), title="NOX Pricing (ILS/ton)")
+    PMx: InterpolatedParam = Field(InterpolatedParam(), title="PMx Pricing (ILS/ton)")
 
-from pydantic import Field, PositiveInt
+    def __init__(self, **data):
+        super().__init__(**data)
 
-from dash_models import DashEditorPage
-from .interpolated_param import InterpolatedParam
+
+class EnergySourceEmissions(DashEditorPage):
+    """
+    The emissions created by an energy sources. Relevant to coal and gas.
+    """
+    CO2: PositiveInt = Field(title="CO2 emissions (g/KW)")
+    SOX: PositiveInt = Field(title="SOX emissions (g/KW)")
+    NOX: PositiveInt = Field(title="NOX emissions (g/KW)")
+    PMx: PositiveInt = Field(title="PMx emissions (g/KW)")
+
+    def __init__(self, **data):
+        super().__init__(**data)
+
+
+class EnergySourceCosts(DashEditorPage):
```

Comment:
For which energy source? Is the intention to have multiple instances of this?

Reply:
yeah, I forgot to handle that. we can just create multiple instances of this with different titles, right?

Reply:
Yes. In that case this should probably inherit from `DashModel` rather than `DashEditorPage` and be an attribute of some encompassing page.

Reply:
right, got it 👍